scripts/west_commands/runners/core.py

```diff
@@ -145,30 +145,49 @@ def _init(self):
         self._parse(self.path)
 
     def _parse(self, filename: str):
-        opt_value = re.compile('^(?P<option>CONFIG_[A-Za-z0-9_]+)=(?P<value>.*)$')
-        not_set = re.compile('^# (?P<option>CONFIG_[A-Za-z0-9_]+) is not set$')
+        # Matches any line, commented out or not, that contains option name
+        # beginning with CONFIG_ add concluded with = followed by, what is
+        # considered, a value or 'is not set' string.
+        # This allows to detect:
+        #   - options commented out by hand: prefix != None and value != None
+        #   - unset options: when prefix != None and value == None
+        #   - enabled/set options: prefix == None and value != None
+        parser = re.compile("^(?P<prefix># )?(?P<option>CONFIG_[a-zA-Z0-9_]+)(=(?P<value>.*)| is not set)$")
 
         with open(filename, 'r') as f:
-            lines = f.readlines()
+            for line in f:
+                match = parser.match(line)
+
+                # Skip commented out lines that are not fitting the pattern
+                if not match:
+                    continue
 
-        for line in lines:
-            match = opt_value.match(line)
-            if match:
                 value = match.group('value')
-                if value.startswith('"') and value.endswith('"'):
-                    value = value[1:-1]
+
+                # Case when line is not commented but value could not be
+                # extracted.
+                if not match.group('prefix') and not value:
+                    _logger.warning("could not extract value from line " + line)
+                    continue
+
+                # If value not set then it is 'n'
+                if not value:
+                    value = 'n'
```

Comment:
Sorry for the delay in reviewing. I'll be watching this PR closely this week.

I still think this is changing the semantics. Before, the only way I see an option `FOO` could be `n` is if the .config file says `# CONFIG_FOO is not set` (or `CONFIG_FOO=n`, which we do in our .conf fragments sometimes, though I think kconfiglib will never emit this).

But here, it looks like both `CONFIG_FOO=` and `# CONFIG_FOO=` result in `FOO` set to `n`.

Can we drop just this patch? It LGTM otherwise.

Reply:
I have reverted to the  original code as the logic to serve both matches with one regex got a little bit absurd.

The change I have added is moving the `for loop`, processing all lines, under the `with` that opens config file as did the original `_parse` code prior to bringing the `load_dot_config` body here.
I have also added `continue` at the end of positive match for `opt_val` to avoid matching the `not_set` as there is no need to check that regex against line in case when the first one matched.